src/plugins/demo/analysisAdapter.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+
+import {Graph} from "../../core/graph";
+import type {RepoId} from "../../core/repoId";
+import type {PluginDeclaration} from "../../analysis/pluginDeclaration";
+import type {IAnalysisAdapter} from "../../analysis/analysisAdapter";
+
+import {declaration} from "./declaration";
+import {graph} from "./graph";
+
+export class AnalysisAdapter implements IAnalysisAdapter {
+  loadingMock: Function;
```

Comment:
Can we type this as

```js
  loadingMock: (assets: Assets, repoId: RepoId) => Promise<mixed>;
``

instead? The `Function` type is deliberately unsound (basically `any`)
and should be avoided.

You can apply the following patch to fix the resulting type error in the test code:

```diff
diff --git a/src/app/adapters/adapterSet.test.js b/src/app/adapters/adapterSet.test.js
index af7c2a78..f0e3caf9 100644
--- a/src/app/adapters/adapterSet.test.js
+++ b/src/app/adapters/adapterSet.test.js
@@ -103,8 +103,7 @@ describe("app/adapters/adapterSet", () => {
     });
     it("loads a dynamicAdapterSet", async () => {
       const {x, sas} = example();
-      x.loadingMock = jest.fn();
-      x.loadingMock.mockResolvedValue();
+      x.loadingMock = jest.fn().mockResolvedValue();
       expect(x.loadingMock).toHaveBeenCalledTimes(0);
       const assets = new Assets("/my/gateway/");
       const repoId = makeRepoId("foo", "bar");
```

Reply:
Yes, for consistency with the app adapter.

Reply:
Okay, makes sense.